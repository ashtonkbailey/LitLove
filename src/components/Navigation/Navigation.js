import React from 'react';
import { Link } from 'react-router-dom';

import '../../index.scss';

const Navigation = (props) => {
  return (
    <div className="nav">
      <Link to="/new">
        <button className="new-search-btn">
          New Search
        </button>
      </Link>
      <Link to="/favorites">
        <button className="favorites-btn">
          Favorites
        </button>
      </Link>
    </div>
  )
}

export default Navigation;
