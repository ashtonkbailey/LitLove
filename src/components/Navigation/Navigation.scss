.nav {
  margin: 3em;
  margin-right: 5em;
  display: flex;
  flex-direction: column;
  justify-content: space-around;

  button {
    background: #DAB192;
    box-shadow:
      0px 4px 4px rgba(0, 0, 0, 0.25),
      inset -5px 6px 15px #EBCBB3;
    width: 9em;
    height: 2em;
    font-family: Sarabun;
    font-weight: bold;
    font-size: 18px;
    text-align: center;
    color: #363636;
    outline: none;
    cursor: pointer;
  }
}