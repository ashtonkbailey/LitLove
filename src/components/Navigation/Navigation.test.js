import React from 'react';
import { shallow } from 'enzyme';

import Navigation from './Navigation';

describe('Navigation', () => {
  let wrapper;

  it('should match snapshot', () => {
    wrapper = shallow(<Navigation />);
    expect(wrapper).toMatchSnapshot();
  });
})
