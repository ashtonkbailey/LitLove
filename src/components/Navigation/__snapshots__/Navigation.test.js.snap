// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navigation />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          replace={false}
          to="/new"
        >
          <button
            className="new-search-btn"
          >
            New Search
          </button>
        </Link>,
        <Link
          replace={false}
          to="/favorites"
        >
          <button
            className="favorites-btn"
          >
            Favorites
          </button>
        </Link>,
      ],
      "className": "nav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button
            className="new-search-btn"
          >
            New Search
          </button>,
          "replace": false,
          "to": "/new",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "New Search",
            "className": "new-search-btn",
          },
          "ref": null,
          "rendered": "New Search",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button
            className="favorites-btn"
          >
            Favorites
          </button>,
          "replace": false,
          "to": "/favorites",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Favorites",
            "className": "favorites-btn",
          },
          "ref": null,
          "rendered": "Favorites",
          "type": "button",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to="/new"
          >
            <button
              className="new-search-btn"
            >
              New Search
            </button>
          </Link>,
          <Link
            replace={false}
            to="/favorites"
          >
            <button
              className="favorites-btn"
            >
              Favorites
            </button>
          </Link>,
        ],
        "className": "nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button
              className="new-search-btn"
            >
              New Search
            </button>,
            "replace": false,
            "to": "/new",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "New Search",
              "className": "new-search-btn",
            },
            "ref": null,
            "rendered": "New Search",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button
              className="favorites-btn"
            >
              Favorites
            </button>,
            "replace": false,
            "to": "/favorites",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Favorites",
              "className": "favorites-btn",
            },
            "ref": null,
            "rendered": "Favorites",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
